### PR TITLE
ENH Allow URLs without trailing slashes

### PR DESCRIPTION
**Pull request by GuySartorelli**
https://github.com/silverstripe/silverstripe-framework/pull/10538 allows us to have a URLs which can be configured to have _or omit_ trailing slashes. This PR updates the campaign-admin javascript to stop explicitly expecting one to be/not be there.

Note that this PR will need to be modified once https://github.com/silverstripe/webpack-config/pull/57 has been merged and released in NPM.

## Parent issue
- silverstripe/silverstripe-cms#2780

### CODE REVIEW

File: client/src/containers/CampaignAdmin/CampaignAdmin.js [hunk 1]
```diff
@@ -79,12 +80,12 @@ class CampaignAdmin extends Component {
...
 
   setBreadcrumbs(view, id, title) {
-    const { sectionConfig: { url } } = this.props;
+    const { sectionConfig: { reactRoutePath } } = this.props;
 
     // Set root breadcrumb
     const breadcrumbs = [{
       text: i18n._t('CampaignAdmin.CAMPAIGN', 'Campaigns'),
-      href: url,
+      href: `/${reactRoutePath}`,
```
Comment (emteknetnz) [inline client/src/containers/CampaignAdmin/CampaignAdmin.js]: Can we always guarantee that reactRoutePath won't have a leading slash?

File: client/src/containers/CampaignAdmin/CampaignAdmin.js [hunk 2]
```diff
@@ -119,7 +120,8 @@ class CampaignAdmin extends Component {
    * @return {string}
    */
   getActionRoute(id, view) {
-    return `/${this.props.sectionConfig.reactRoutePath}/set/${id}/${view}`;
+    const { reactRoutePath } = this.props.sectionConfig;
+    return joinUrlPaths(`/${reactRoutePath}`, `/set/${id}/${view}`);
```
Comment (emteknetnz) [inline client/src/containers/CampaignAdmin/CampaignAdmin.js]: This is possibly better done as 
```    return joinUrlPaths('/', reactRoutePath, `/set/${id}/${view}`);```

I'm not sure if reactRouthPath could have a leading slash or not

Same applies for all the other cases of ````/${reactRoutePath}```` in this PR

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: client/src/containers/CampaignAdmin/CampaignAdmin.js (hunk 1 above)
Comment (GuySartorelli) [reply client/src/containers/CampaignAdmin/CampaignAdmin.js]: Theoretically yes but realistically there's the remote chance someone will do something really weird lol. I'll apply `joinUrlPaths` to it as you recommended in another comment, that's a really elegant way to handle it that I hadn't considered

File: client/src/containers/CampaignAdmin/CampaignAdmin.js (hunk 2 above)
Comment (GuySartorelli) [reply client/src/containers/CampaignAdmin/CampaignAdmin.js]: Done

File: client/src/containers/CampaignAdmin/CampaignAdmin.js (hunk 1 above)
Comment (GuySartorelli) [reply client/src/containers/CampaignAdmin/CampaignAdmin.js]: Done

**Comment by GuySartorelli** [comment]
webpack-config changes are now included in this PR.

**Comment by GuySartorelli** [comment]
Rebased to resolve conflicts

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**